### PR TITLE
feat(github): releases, gists(graphql)

### PR DESCRIPTION
**Pull request by shreyas1599**
- Gists (graphql)
- Releases (download url is the first downloadable asset)
- Graphql response does not return source code assets:
in the below image only the apk asset is returned. 
image

- ~Possible Improvement: If a release has multiple non-source code assets then use SimpleDialog, but there is no equivalent cupertino widget. Not sure if we should create a separate screen just for this?~

- Screenshot
image

### CODE REVIEW

File: lib/widgets/release_item.dart [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      @required this.downloadUrl,
+      @required this.name,
+      @required this.tagName,
+      @required this.avatarUrl});
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = Provider.of<ThemeModel>(context);
+    return Link(
+      url: downloadUrl,
```
Comment (pd4d10) [inline lib/widgets/release_item.dart:29]: Would it be better to go to the release detail web page? For example: https://github.com/git-touch/git-touch/releases/tag/v1.10.1

File: lib/screens/gh_repo.dart [hunk 2]
```diff
@@ -333,7 +333,12 @@ class GhRepoScreen extends StatelessWidget {
                       },
                     ),
                     url: '/github/$owner/$name/contributors',
-                  )
+                  ),
+                  TableViewItem(
+                    leftIconData: Octicons.book,
+                    text: Text("Releases"),
+                    url: '/github/$owner/$name/releases',
+                  ),
```
Comment (pd4d10) [inline lib/screens/gh_repo.dart:341]: 1. We could have releases count here.
2. The releases entry in the action sheet (top-right menu) could be removed

File: lib/graphql/github.graphql [hunk 3]
```diff
@@ -1018,3 +1062,31 @@ query Issue($owner: String!, $name: String!, $number: Int!, $cursor: String) {
     }
   }
 }
+
+# Releases
+query Releases($name: String!, $owner: String!, $cursor: String) {
+   repository(name: $name, owner: $owner) {
+    releases(first: 30, after: $cursor) {
```
Comment (pd4d10) [inline lib/graphql/github.graphql:1069]: Could we set the order? DESC by time would be fine, just the same as the web releases page

**Comment by pd4d10** [comment]
Thanks!

I'm thinking if we should add more information to the releases page. This is the mobile web version:



simulator_screenshot_E9D7D45E-AABD-4814-9002-A630C9EC071F

**Comment by shreyas1599** [comment]
I thought about this. There's a `description` prop for release but it does not seem to output what I thought it would (the "bug fixes" text in the image you included). It returns an empty string. Not sure if it's a bug or if it means something else. I've included all useful information exposed through the connection(except releasesCount. I'll add that). It's only there in 'REST`. [Example url](https://api.github.com/repos/git-touch/git-touch/releases). Even the source code zip and tarball files are in here. Should we change it to rest or wait for github to expose these through graphql? I'll file an issue in github.community?

File: lib/widgets/release_item.dart (hunk 1 above)
Comment (shreyas1599) [reply lib/widgets/release_item.dart:29]: Alright. Sounds better until we can display all the assets so the user knows what they're downloading. Or should I use `ListTile` which expands to display the assets and then use downloadUrl for that?

File: lib/screens/gh_repo.dart (hunk 2 above)
Comment (shreyas1599) [reply lib/screens/gh_repo.dart:341]: I noticed there's projects also. I'll add a screen for that as well.

**Comment by pd4d10** [comment]
> change it to rest or wait for github to expose these through graphql

I think both would be fine. At least we could use the web version

> file an issue in github.community

It's a good idea. But I'm not sure if it is in the official plan

**Comment by shreyas1599** [comment]
> There's a `description` prop for release but it does not seem to output what I thought it would (the "bug fixes" text in the image you included). It returns an empty string. Not sure if it's a bug or if it means something else.

Sorry my bad. the description returns a proper response. I was looking at earlier releases which had no descriptions. I ordered by descending order and the description works as expected.

**Comment by shreyas1599** [comment]
Hi @pd4d10, I made a few changes. I thought it'd be better to list the assets and add a download button within the app itself instead of redirecting) Looks like this now:

Screenshot_1612089261

File: lib/widgets/release_item.dart [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+            description,
+          ),
+          SizedBox(height: 10),
+        ],
+        ExpansionTile(
+          title: Text(
+            'Assets (' +
+                (releaseAssets != null
+                    ? releaseAssets.nodes.length.toString()
+                    : '0') +
```
Comment (pd4d10) [inline lib/widgets/release_item.dart:82]: Conditional property access could help in this case:

```dart
(releaseAssets?.nodes?.length ?? 0).toString()
```

**Comment by pd4d10** [comment]
The lint is fixed in the latest `main` branch. Could you do a rebase?

**Review by pd4d10 — APPROVED** [review]

**Pull request merged**